### ISSUE TITLE
Should frobenius_map powers be unrolled

### ISSUE THREAD

**Issue by ValarDragon** [arkworks-rs/algebra#209]
Currently we have a method `fe.frobenius_map(power)`, which computes a given power of a frobenius map. This involves multiplying a field element by frobenius coefficient. In many instances, these frobenius coefficients are `1, -1`, however we resort to general multiplication in these cases.

The frobenius map power is almost always known at compile time, (and as far as I can tell only ranges between 0,1,2,3 in usage within the library), so we could switch this for a potential #206 optimization if we change the API to have different functions for different frobenius powers, or have a macro of some sort for computing this when the power is known at compile time.

**Comment by Pratyush** [comment]
This is a good point, but I dunno how much we can save, because we only call the frobenius map a few times.

**Comment by ValarDragon** [comment]
These are known at compile time, so theres no run time test needed. (The coefficient is a constant)

**Comment by ValarDragon** [comment]
Agreed that its probably not that notable on a second pass through, its called like 4 times for bls12
